src/auth/Config.h

```diff
@@ -26,7 +26,7 @@ class Config
 public:
     Config() = default;
     explicit Config(Config &&) = default;
```

Comment:
This PR makes this move constructor (and any implicit copy methods) especially buggy. Please prohibit Auth::Config copy/move methods where possible (the untested suggestion below prohibits all of them) and explicitly define any remaining ones. Hopefully, after the prohibition suggested below, there will not be any left to define manually!

```suggestion
    Config(Config &&) = delete; // no support for copying of any kind
```